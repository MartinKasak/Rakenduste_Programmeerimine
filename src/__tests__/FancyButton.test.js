import React from "react";
import renderer from "react-test-renderer";
import FancyButton from "../components/FancyButton";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

describe("FancyButton component", () =>{
  test("snapshot", () =>{
    const component = renderer.create(
      <FancyButton onClick={() => 0}>
        Hello
      </FancyButton>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

it("should trigger callback when the button is clicked", () =>{
    let x = 0;
    const component = shallow(
      <FancyButton onClick={() => x = x+1}>
        Hello
      </FancyButton>
    );
    component.find(".btn").simulate("click");
    expect(x).toBe(1);
  });