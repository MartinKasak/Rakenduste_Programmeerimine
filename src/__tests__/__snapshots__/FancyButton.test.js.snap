// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FancyButton component snapshot 1`] = `
<div
  className="btn btn--fancy"
  onClick={[Function]}
>
  <div
    className="btn-inner"
  >
    <div>
      Hello
    </div>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </div>
</div>
`;
